drivers/flash/flash_stm32_ospi.c

```diff
@@ -903,9 +903,10 @@ static int flash_stm32_ospi_read(const struct device *dev, off_t addr,
 			cmd.Instruction = SPI_NOR_OCMD_RD;
 			cmd.DummyCycles = SPI_NOR_DUMMY_RD_OCTAL;
 		} else {
-			/* use SFDP:BFP read instruction */
-			cmd.Instruction = dev_data->read_opcode;
-			cmd.DummyCycles = dev_data->read_dummy;
+			/* SPI and STR : use fast read with addr on 4 bytes */
+			cmd.Instruction = SPI_NOR_CMD_READ_FAST_4B;
+			cmd.DummyCycles = SPI_NOR_DUMMY_RD;
```

Comment:
Why use hardcoded read opcode/instruction here?
Not every flash supports `FAST_4B` (_e.g. some winbond_) and the opcode can be discovered from SFDP...

If we have to hardcode that, then I propose to revert this block back to the SFDP approach but allow to set custom read opcode in the DTS. Otherwise, that change breaks some flash ICs.

Reply:
Ok restoring the instruction set to dev_data->read_opcode with a comment to remind that SPI_NOR_CMD_READ_FAST_4B is prefered

Reply:
> Ok restoring the instruction set to dev_data->read_opcode with a comment to remind that SPI_NOR_CMD_READ_FAST_4B is prefered

Ok. I don't think that there is any preferred read opcode as it depends entirely on the flash and/or manufacturer.
Maybe we shall add `readoc` into DTS and prioritize it over SFDP?

Reply:
Yes, but maybe in _a next PR,_ to fix the initial issue now.
we could add a property in the st,stm32-ospi-nor.yaml like: 
```
    readoc:
      type: int
      required: false
      enum:
        - "READ"  		# 0x03   SPI_NOR_CMD_READ /* Read data */
        - "READ_4B" 		# 0x13   SPI_NOR_CMD_READ_4B  /* Read data 4 Byte Address */
        - "READ_FAST_4B" 	# 0x0C   SPI_NOR_CMD_READ_FAST_4B  /* Fast Read 4 Byte Address */
        - "OCTAL_RD"		# 0xEC13 SPI_NOR_CMD_OCTAL_RD   /* octal Read data */
        - "DREAD" 		# 0x3B SPI_NOR_CMD_DREAD /* Read data (1-1-2) */
        - "2READ" 		# 0xBB SPI_NOR_CMD_2READ /* Read data (1-2-2) */
        - "QREAD" 		# 0x6B SPI_NOR_CMD_QREAD /* Read data (1-1-4) */
        - "4READ" 		# 0xEB SPI_NOR_CMD_4READ /* Read data (1-4-4) */
      description: |
        The value is the instruction to send to the octoflash for read operations.
        If present it will overwrites the value from the SFDP table parameter
```
and then use this property when reading (flash_stm32_ospi_read) instead of value from the SFDP : 
```
#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_ospi_nor), readoc)
			/* take the instruction from the DTS */
			cmd.Instruction = DT_INST_PROP(0, readoc),
#else
			/* use SFDP:BFP read instruction */
			cmd.Instruction = dev_data->read_opcode;
#endif

```

Reply:
@GeorgeCGV I agree with @FRASTM. Rather than trying to fix every potential failure we can meet in current PR, we should focus to fix initial regression (https://github.com/zephyrproject-rtos/zephyr/issues/46740), which is blocking multiple activities (see https://github.com/zephyrproject-rtos/zephyr/issues/48175 for instance).
Then we can deal with other potential cases in a new PR.

Reply:
@FRASTM, yes `readoc` could be a good workaround for now.
@erwango, sure as there are more OSPI PRs piling up.